rpc/dispatch.go

```diff
@@ -78,13 +87,13 @@ func (d *dispatch) Call(ctx context.Context, name string, arg interface{}, res i
 	}
 }
 
-func (d *dispatch) Notify(ctx context.Context, name string, arg interface{}) error {
+func (d *dispatch) Notify(ctx context.Context, name string, arg interface{}, sendNotifier SendNotifier) error {
 	rpcTags, _ := RpcTagsFromContext(ctx)
 	v := []interface{}{MethodNotify, name, arg}
 	if len(rpcTags) > 0 {
 		v = append(v, rpcTags)
 	}
-	errCh := d.writer.EncodeAndWrite(ctx, v)
+	errCh := d.writer.EncodeAndWrite(ctx, v, currySendNotifier(sendNotifier, SeqNumber(0)))
```

Comment:
Might be clearer to use `SeqNumber(-1)` here. Since `-1` is impossible in the happy path, it acts as a better placeholder or sentinel value.